src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs

```diff
@@ -115,24 +115,43 @@ protected virtual void OnScriptFileChanged(object sender, FileSystemEventArgs e)
         /// </summary>
         protected static void ApplyAmbientBindingData(IBinder binder, IDictionary<string, string> bindingData)
         {
-            // TEMP: Dig the ambient binding data out of the binder
-            FieldInfo fieldInfo = binder.GetType().GetField("_bindingSource", BindingFlags.NonPublic | BindingFlags.Instance);
-            var bindingSource = fieldInfo.GetValue(binder);
-            PropertyInfo propertyInfo = bindingSource.GetType().GetProperty("AmbientBindingContext");
-            var ambientBindingContext = propertyInfo.GetValue(bindingSource);
-            propertyInfo = ambientBindingContext.GetType().GetProperty("BindingData");
-            IDictionary<string, object> ambientBindingData = (IDictionary<string, object>)propertyInfo.GetValue(ambientBindingContext);
-
+            var ambientBindingData = GetAmbientBindingData(binder);
             if (ambientBindingData != null)
             {
                 // apply the binding data to ours
                 foreach (var item in ambientBindingData)
                 {
-                    bindingData[item.Key] = item.Value.ToString();
+                    if (item.Value != null)
```

Comment:
This null check is the fix. All the other changes are just me being more defensive.